### PR TITLE
Add php-7.4 support

### PR DESCRIPTION
**Pull request by samanthajayasinghe**

### CODE REVIEW

File: compose/php.yml [hunk 1]
```diff
@@ -295,5 +295,47 @@ services:
...
+    networks:
+      - totara
+
+  php-7.4-debug:
+    image: totara/docker-dev-php74-debug
+    container_name: totara_php74_debug
+    working_dir: ${REMOTE_SRC}
+    environment:
+      XDEBUG_CONFIG: remote_host=${HOST_IP}
+      PHP_IDE_CONFIG: serverName=totara73
```
Comment (derschatta) [inline compose/php.yml]: this should be changed to totara74

File: php/php74-debug/Dockerfile [hunk 2]
```diff
@@ -0,0 +1 @@
+FROM totara/docker-dev-php74:latest
```
Comment (derschatta) [inline php/php74-debug/Dockerfile]: I also think you need to change xdebug to version 2.8.0beta2 which adds PHP 7.4 support

File: php/php74/config/php.ini [hunk 3]
```diff
@@ -0,0 +1,18 @@
+log_errors = On
+error_log = /dev/stderr
+error_reporting = E_ALL & ~E_DEPRECATED
+max_input_vars = 2000
+date.timezone = Pacific/Auckland
+memory_limit=256M
+opcache.optimization_level=0x7FFFBBFF
+; necessary as Xdebug impacts performance that hard
+max_execution_time=120
+;opcache.revalidate_freq=0
```
Comment (derschatta) [inline php/php74/config/php.ini]: I think you can remove those commented opcache values

File: php/php74/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    mssql-tools \
+    unixodbc-dev
+
+# Workaround to get MSSQL connection working on Debian 9 (stretch)
+# https://emacstragic.net/2017/11/06/mssql-odbc-client-on-debian-9-stretch/
+RUN wget "http://security.debian.org/debian-security/pool/updates/main/o/openssl/libssl1.0.0_1.0.1t-1+deb8u11_amd64.deb" \
+    && DEBIAN_FRONTEND=noninteractive dpkg -i ./libssl1.0.0_1.0.1t-1+deb8u11_amd64.deb
+
+RUN echo 'export PATH="$PATH:/opt/mssql-tools/bin"' >> ~/.bash_profile \
+    && echo 'export PATH="$PATH:/opt/mssql-tools/bin"' >> ~/.bashrc
```
Comment (derschatta) [inline php/php74/Dockerfile]: after that add the following to get mssql support working:
```
RUN pear config-set php_ini `php --ini | grep "Loaded Configuration" | sed -e "s|.*:\s*||"` system \
    && pecl install sqlsrv-5.7.0preview \
    && pecl install pdo_sqlsrv-5.7.0preview

RUN docker-php-ext-enable sqlsrv.so && docker-php-ext-enable pdo_sqlsrv.so
```

**Comment by derschatta** [comment]
Tested it together with PostgreSQL and Mssql, works. Still might need to double check mariadb but I don't expect a problem.

File: compose/php.yml (hunk 1 above)
Comment (samanthajayasinghe) [reply compose/php.yml]: done

File: php/php74-debug/Dockerfile (hunk 2 above)
Comment (samanthajayasinghe) [reply php/php74-debug/Dockerfile]: done

File: php/php74/config/php.ini (hunk 3 above)
Comment (samanthajayasinghe) [reply php/php74/config/php.ini]: done

File: php/php74/Dockerfile (hunk 4 above)
Comment (samanthajayasinghe) [reply php/php74/Dockerfile]: done

**Comment by samanthajayasinghe** [comment]
@derschatta I fixed most of the issues .. pls have another look

**Comment by derschatta** [comment]
> @derschatta I fixed most of the issues .. pls have another look

Look goods, xdebug works, mssql and mariadb also.

**Comment by derschatta** [comment]
@samanthajayasinghe I also created the repositories on docker hub

**Pull request merged**